avm/res/security-insights/setting/main.bicep

```
@@ -3,6 +3,11 @@ metadata description = 'This module deploys a Security Insights Setting.'
 metadata owner = 'Azure/module-maintainers'
 
 import { roleAssignmentType } from 'br/public:avm/utl/types/avm-common-types:0.5.1'
+@description('Optional. Array of role assignments to create.')
+param roleAssignments roleAssignmentType[]?
+
+@description('Optional. Tags of the resource.')
+param tags object?
```

```suggestion
param tags resourceInput<'Microsoft.SecurityInsights/settings@2024-10-01-preview'>.tags?
```

Small improvement. :) Requires a new build of the template though

---

Updated

---

You sure? 😄 Please check if a push is missing.

---

Actually, this throws an error when adding, so I don't believe tags are supported.

---

you bet they are 😄 But you have to use the latest Bicep version `0.35.1` ⚠️

---

But if what you wrote in the other thread is true, the issue is not the `resourceInput` but that tags themselves are not supported

---

Adding this line and then updating the WAF test, I get the following errors:

/Users/matthewschmitt/GIT/bicep-registry-modules/avm/res/security-insights/setting/main.bicep(10,84) : Error BCP055: Cannot access properties of type "Microsoft.SecurityInsights/settings". An "object" type is required. [https://aka.ms/bicep/core-diagnostics#BCP055]
/Users/matthewschmitt/GIT/bicep-registry-modules/avm/res/security-insights/setting/main.bicep(80,9) : Error BCP062: The referenced declaration with name "tags" is not valid. [https://aka.ms/bicep/core-diagnostics#BCP062]
/Users/matthewschmitt/GIT/bicep-registry-modules/avm/res/security-insights/setting/tests/e2e/max/main.test.bicep(48,23) : Error BCP104: The referenced module has errors. [https://aka.ms/bicep/core-diagnostics#BCP104]
/Users/matthewschmitt/GIT/bicep-registry-modules/avm/res/security-insights/setting/tests/e2e/max/main.test.bicep(52,5) : Error BCP035: The specified "object" declaration is missing the following required properties: "tags". [https://aka.ms/bicep/core-diagnostics#BCP035]

Which lead me to believe that tags is not supported on this resource.

---

Agreed. Also tags are not documented : https://learn.microsoft.com/en-us/azure/templates/microsoft.securityinsights/settings?pivots=deployment-language-bicep

Thanks for checking :)